crates/bevy_math/src/bezier.rs

```diff
@@ -0,0 +1,253 @@
+/// to ensure the animation begins at 0, and ends at 1.
+#[derive(Default, Clone, Copy, Debug, PartialEq)]
+pub struct CubicBezierEasing {
+    /// Control point P1 of the 2D cubic Bezier curve. Controls the start of the animation.
+    p1: Vec2,
+    /// Control point P2 of the 2D cubic Bezier curve. Controls the end of the animation.
+    p2: Vec2,
+}
+
+impl CubicBezierEasing {
+    /// Construct a cubic bezier curve for animation easing, with control points `p1` and `p2`.
+    pub fn new(p1: Vec2, p2: Vec2) -> Self {
+        Self { p1, p2 }
+    }
+
+    /// Maximum allowable error for iterative bezier solve
+    const MAX_ERROR: f32 = 1e-7;
+
+    /// Maximum number of iterations during bezier solve
+    const MAX_ITERS: u8 = 16;
+
+    /// Given a `time` within `0..=1`, remaps to a new value using the cubic Bezier curve as a
+    /// shaping function, for which when plotted `x = time` and `y = animation progress`. This will
+    /// return `0` when `t = 0`, and `1` when `t = 1`.
+    pub fn ease(&self, time: f32) -> f32 {
+        let x = time.clamp(0.0, 1.0);
+        let t = self.find_t_given_x(x);
+        self.evaluate_y_at(t)
+    }
+
+    /// Compute the x-coordinate of the point along the Bezier curve at `t`.
+    #[inline]
+    pub fn evaluate_x_at(&self, t: f32) -> f32 {
+        bezier_impl::position([0.0, self.p1.x, self.p2.x, 1.0], t)
+    }
+
+    /// Compute the y-coordinate of the point along the Bezier curve at `t`.
+    #[inline]
+    pub fn evaluate_y_at(&self, t: f32) -> f32 {
+        bezier_impl::position([0.0, self.p1.y, self.p2.y, 1.0], t)
+    }
+
+    /// Compute the slope of the line at the given parametric value `t` with respect to t.
+    #[inline]
+    pub fn dx_dt(&self, t: f32) -> f32 {
+        bezier_impl::velocity([0.0, self.p1.x, self.p2.x, 1.0], t)
+    }
+
+    /// Solve for the parametric value `t` that corresponds to the given value of `x`.
+    #[inline]
+    pub fn find_t_given_x(&self, x: f32) -> f32 {
+        let mut t_guess = x;
+        (0..Self::MAX_ITERS).any(|_| {
+            let x_guess = self.evaluate_x_at(t_guess);
+            let error = x_guess - x;
+            if error.abs() <= Self::MAX_ERROR {
+                true
+            } else {
+                let slope = self.dx_dt(t_guess);
+                t_guess -= error / slope;
+                false
+            }
+        });
+        t_guess.clamp(0.0, 1.0)
+    }
+}
+
+/// Generic implementations for sampling cubic Bezier curves. Consider using the methods on
+/// [`Bezier`] for more ergonomic use.
+pub mod bezier_impl {
+    use super::Point;
+
+    /// Compute the bernstein basis polynomial for iteration `i`, for a Bezier curve with with
+    /// degree `n`, at `t`.
+    #[inline]
+    pub fn bernstein_basis(n: usize, i: usize, t: f32) -> f32 {
+        (1. - t).powi((n - i) as i32) * t.powi(i as i32)
+    }
+
+    /// Efficiently compute the binomial coefficient
```

Comment:
```suggestion
    /// Efficiently compute the binomial coefficient of `n` choose `k`. 
```

Reply:
Ah, is that how you say it in plain English? This was my first time running into binomial coefficient notation, and initially I had no idea I was supposed to evaluate it down to a value.